0010-release-strategy.adoc

```diff
@@ -0,0 +1,505 @@
+
+
+== Release Trigger Guidelines - Recommendation
+
+This is an attempt to consider and review what could be the reason to trigger a release.
+Inline is the author's recommendation
+
+[cols="1,1,1,2"]
+|===
+| Repo (comit-?)| Description | Triggers a release? | Reason
+
+| i/rs | Security bug fix | Yes | We don't want users to lose funds
+| i/rs | Feature that resolves a quaterly KR | Yes | Mark the achievement and consider it done
+| i/rs | Code refactoring | No | Does not bring value to the user
+| i/rs | Test improvement | No | Does not bring value to the user
+| i/rs | UX Feature | Yes | As it bring value to the user
+| rs | Breaking HTTP API Change | Yes | To force us to (hopefully) align the embedded comit-i as part of the release validation process
+| i | Adapt to comit-rs HTTP API Change | Yes | To make it easier to work on comit-rs with the embedded comit-i
+| rs | Breaking COMIT API Change | Yes | To migrate the "network" to the latest API fast and reduce the number of user using a deprecated API; To be able to easily differentiate both protocols
+| rs | Internal API (btsieve REST API) | No | Does not bring value to the user
+| rs | CLI API | Depends | Whether it fits under the _UX Feature_ category
+| rs | Embedded comit-i | Depends | Whether the comit-i changes fits in any other categories above
+| i/rs | Any other change | Soft No | A new feature that does not fit in any category above should not trigger a release, except if the team think it should (ie, ad-hoc discussion)
+
+|===
+
+Author note: Let me know if I forgot something
+
+== When to increment major/minor/patch number - Recommendation
+
+If a release contains several changes then we should increment the heaviest number (with patch < minor < major).
+The list below only contains elements from section <<Release Trigger Guidelines - Recommendation>> (because you don't need to increment the version if you don't do a release for such change).
+
+
+[cols="1,1,1,1,2"]
+|===
+| repo (comit-?) | Change | Pre-1.0.0 | Post-1.0.0 | Comment
+
+| i/rs | Security bug fix | Patch | Patch | As per standard guidelines
+| i/rs | Feature that resolves a quaterly KR | Minor | Minor | Except if it fits in another category
+| i/rs | UX Feature | Minor | Major/Minor | Team decides depending how ground-breaking the feature is, e.g, how much users will have to re-learn to use COMIT
+| i/rs | Breaking HTTP API Change | Minor | Major | If someone were to create a client on cnd, they need to know that they can upgrade minor versions without risk
+| i | Adapt to comit-rs HTTP API Change | Minor | Major |
+| rs | Breaking COMIT API Change | Minor | Major | To express non-backward compatibility between two cnd
+| rs | Breaking CLI API | Minor | Major | While this is unlikely to happen, you don't want user to discover that their systemd scripts are broken by surprise
+| rs | Embedded comit-i | Minor | Major/Minor | Depending on whether the comit-i changes fits in any other categories above
+
```

Comment:
Perhaps a row for Rust 'breaking changes' as defined in by the Cargo docs (quoted in comment above).

Reply:
Same than before, I don't think it applies here as it's a binary crate.